de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/comments/CommentUtil.java

```diff
@@ -29,12 +31,24 @@
 	private CommentUtil() {
 	}
 
-	public static boolean shapeBelongsToFaultNode(Shape shape, FaultTreeNode bean) {
+	/**
+	 * @param shape
+	 * @param bean business object
+	 * @return true if shape belongs to specified business object
+	 */
+	public static boolean shapeBelongsToEntity(Shape shape, Object bean) {
+		String uuid = null;
 		String propertyValue = Graphiti.getPeService().getPropertyValue(shape, BELONGS_TO_FAULT_NODE);
 		if (propertyValue == null) {
 			return false;
 		}
-		return propertyValue.equals(bean.getUuid());
+
+		if (bean instanceof FaultTreeNode) {
```

Comment:
I think theres an interface IBeanUUID that we can use here instead of just "Object". Then we dont need to do this if check here.